tests/python/gaia-ui-tests/gaiatest/tests/graphics/test_marketplace_download_execute_map_app.py

```diff
@@ -34,62 +34,48 @@ def test_install_and_execute_bing_map(self):
 
         results = marketplace.search(self.app_search)
         first_result = results.search_results[0]
-        app_name = first_result.get_app_name()
         first_result.tap_install_button()
 
-        # Confirm the installation and wait for the app icon to be present
+        # Confirm the installation
         confirm_install = ConfirmInstall(self.marionette)
         confirm_install.tap_confirm()
 
-        self.assertEqual(self.apps.displayed_app.name, 'Marketplace')
-
-        self.device.touch_home_button()
-
-        # Check that the icon of the app is on the homescreen
-        homescreen = Homescreen(self.marionette)
-        homescreen.wait_for_app_icon_present(app_name)
-
-        installed_app = homescreen.installed_app(app_name)
-        installed_app.tap_icon()
+        # Do a search again and get the element again.  Since first_result is a pageregion object,
+        # after the frame change done during the install the element will become stale and need to be recaptured.
+        results = marketplace.search(self.app_search)
```

Comment:
Perhaps you could do results = SearchResults(self.marionette) here and work from there?
It seems a shame that you have to reload the page just because of this bug.
I think you should also mention the issue that is going on here, namely https://bugzilla.mozilla.org/show_bug.cgi?id=1202246